cnd/src/btsieve/ethereum/mod.rs

```diff
@@ -222,51 +222,55 @@ where
     for transaction in block.transactions.into_iter() {
         let tx_hash = transaction.hash;
 
-        let receipt = if needs_receipt {
-            let receipt = connector
-                .receipt_by_hash(tx_hash)
-                .compat()
-                .await?
-                .ok_or_else(|| {
-                    anyhow::anyhow!(
-                        "Could not get transaction receipt for transaction {:x}",
-                        tx_hash
-                    )
-                })?;
-
-            Some(receipt)
-        } else {
-            None
-        };
-
-        let result = pattern.matches(&transaction, receipt.as_ref());
+        let receipt = connector
+            .receipt_by_hash(tx_hash)
+            .compat()
+            .await?
+            .ok_or_else(|| {
+                anyhow::anyhow!(
+                    "Could not get transaction receipt for transaction {:x}",
+                    tx_hash
+                )
+            })?;
 
         tracing::debug!(
-            "matching {:?} against transaction {:x} yielded {}",
-            pattern,
+            "checking transaction {:x} status: {:?}",
             tx_hash,
-            result
+            receipt.status
         );
 
-        if result {
-            let receipt = match receipt {
-                Some(receipt) => receipt,
-                None => connector
-                    .receipt_by_hash(tx_hash)
-                    .compat()
-                    .await?
-                    .ok_or_else(|| {
-                        anyhow::anyhow!(
-                            "Could not get transaction receipt for transaction {:x}",
-                            tx_hash
-                        )
-                    })?,
-            };
-
-            return Ok(Some(TransactionAndReceipt {
-                transaction,
-                receipt,
-            }));
+        if receipt.status == Some(1.into()) {
+            let receipt = if needs_receipt { Some(receipt) } else { None };
+
+            let result = pattern.matches(&transaction, receipt.as_ref());
+
+            tracing::debug!(
+                "matching {:?} against transaction {:x} yielded {}",
+                pattern,
+                tx_hash,
+                result
+            );
+
+            if result {
+                let receipt = match receipt {
+                    Some(receipt) => receipt,
+                    None => connector
+                        .receipt_by_hash(tx_hash)
```

Comment:
hmmm, I think this is broken mate.  We fetch the receipt by hash again here for the same `tx_hash` as above?

Reply:
Note that this code was already here, we already used to fetch the receipt lazily twice (well, there are 2 blocks fetching it but only one would fetch it).